workers/unity/Packages/com.improbable.gdk.health/Systems/ServerHealthModifierSystem.cs

```diff
@@ -9,54 +8,57 @@ namespace Improbable.Gdk.Health
     [UpdateInGroup(typeof(SpatialOSUpdateGroup))]
     public class ServerHealthModifierSystem : ComponentSystem
     {
-        private struct EntitiesWithModifiedHealth
+        private WorkerSystem workerSystem;
+        private CommandSystem commandSystem;
+        private ComponentUpdateSystem componentUpdateSystem;
+
+        protected override void OnCreateManager()
         {
-            public readonly int Length;
-            [ReadOnly] public EntityArray Entity;
-            public ComponentDataArray<HealthComponent.Component> Health;
-            [ReadOnly] public ComponentDataArray<HealthComponent.CommandRequests.ModifyHealth> ModifyHealthRequests;
-            public ComponentDataArray<HealthComponent.EventSender.HealthModified> HealthModifiedEventSenders;
-        }
+            base.OnCreateManager();
 
-        [Inject] private EntitiesWithModifiedHealth entitiesWithModifiedHealth;
+            workerSystem = World.GetExistingManager<WorkerSystem>();
+            commandSystem = World.GetExistingManager<CommandSystem>();
+            componentUpdateSystem = World.GetExistingManager<ComponentUpdateSystem>();
+        }
 
         protected override void OnUpdate()
         {
-            for (var i = 0; i < entitiesWithModifiedHealth.Length; i++)
+            var requests = commandSystem.GetRequests<HealthComponent.ModifyHealth.ReceivedRequest>();
+            var healthComponentData = GetComponentDataFromEntity<HealthComponent.Component>();
```

Comment:
Can skip this if there are no requests to process

Reply:
👍